Framework/PythonInterface/plugins/algorithms/BASISDiffraction.py

```diff
@@ -487,10 +487,12 @@ def _spawn_tempnexus(self):
         Create a temporary file and flag for removal upon algorithm completion.
         :return: (str) absolute path to the temporary file.
         """
-        file_object, file_name = \
-            tempfile.mkstemp(prefix='BASISDiffraction_',
-                             suffix='.nxs',
-                             dir=mantid_config['defaultsave.directory'])
+        f = tempfile.NamedTemporaryFile(prefix='BASISDiffraction_',
+                                        suffix='.nxs',
+                                        dir=mantid_config['defaultsave.directory'],
+                                        delete=False)
+        file_name = f.name
+        f.close()
```

Comment:
Code changes look fine. I ran the test on Ubuntu 16.04 but someone with experience building Mantid on Windows should check too.